Feature/ignore mouse events

montegoulding opened the pull request:
This new stack property allows a stack to ignore mouse events so the events pass through the stack or the stack is transparent to the mouse events. It's implemented for OS X and Windows only so far. I have forked develop for this work because it is dependant on the Cocoa implementation in 6.7. If at all possible I'd appreciate this property being integrated into the 6.7.0 release cycle.

Commit:
[[ IgnoreMouseEvents ]] Implemented the ignoreMouseEvents property which will make a stack transparent to mouse events.

Commit:
[[ IgnoreMouseEvents ]] Implemented on Windows using WS_EX_TRANSPARENT rather than HTTRANSPARENT to match the behavior on OS X

Commit:
[[ IgnoreMouseEvents ]] Adding documentation and implementation note

Commit:
Merge branch 'develop' into feature/ignoreMouseEvents

Conflicts:
	engine/src/stack.cpp

runrevmark commented:
Hi Monte - thanks for this, I've merged this into develop and its just in time for 6.7-dp-5 which Michael is building this morning.

montegoulding commented:
OK, I have removed the offending line. If you pull this branch from me again Mark you will see a new commit which resolves the issue. For some reason setting GWL_STYLE made the window disappear here and I added that when I found `getstyle` to be an appropriate place to move some code to and it set both GLW_STYLE and GWL_EXSTYLE but I skipped a final test for some reason. Sorry about that.

runrevmark commented:
That update has been merged into 'develop'.

Thanks Monte.
